### PR TITLE
Update default anvil layouts for S and M sizes of ne30pg2_EC30to60E2r2

### PR DESCRIPTION
**Pull request by jonbob**
One of our users, @alicebarthel, was doing a series of ne30pg2_EC30to60E2r2 runs on anvil and the PACE output showed the S and M default layouts could be further optimized. This PR updates these default layouts with some improvement without changing the overall processor counts. S goes from ~5.0 to 5.4 sypd, and M improves from ~8.8 to 9.4 sypd.

[BFB]

### CODE REVIEW

**Comment by alicebarthel** [comment]
Reading through the commits, the changes look consistent with the improvements we had talked about when looking at the performance of the prior layouts. 
@jonbob I suppose the performance for the new layouts are [here](https://pace.ornl.gov/exp-details/101795) (for S) and [here](https://pace.ornl.gov/exp-details/101800) (for M)?

**Review by alicebarthel — APPROVED** [review]

**Comment by jonbob** [comment]
@alicebarthel -- thank you for pointing to that PACE output. They're exactly the tests I had made and used for this PR. Please note they don't have all the white space that your runs did when we looked at them.

**Comment by alicebarthel** [comment]
Thanks @jonbob! That is exactly what I looked out for once I found the PACE links. Happy to see the improved throughput :)

**Comment by jonbob** [comment]
passes sanity testing, merged to next

**Comment by jonbob** [comment]
merged to master and expected NML DIFFs (due to layouts) blessed

File: cime_config/allactive/config_pesall.xml [hunk 1]
```diff
@@ -6958,22 +6958,22 @@
...
       <pes compset=".*EAM.+ELM.+MPASSI.+MPASO.+MOSART.+" pesize="M">
-        <comment> --compset WCYCL* --res ne30pg2_EC30to60E2r2 on 48 nodes pure-MPI, ~8.8 sypd </comment>
+        <comment> --compset WCYCL* --res ne30pg2_EC30to60E2r2 on 48 nodes pure-MPI, ~9.4 sypd </comment>
         <ntasks>
           <ntasks_atm>1350</ntasks_atm>
-          <ntasks_lnd>288</ntasks_lnd>
-          <ntasks_rof>288</ntasks_rof>
-          <ntasks_ice>1080</ntasks_ice>
+          <ntasks_lnd>72</ntasks_lnd>
+          <ntasks_rof>72</ntasks_rof>
```
Comment (amametjanov) [inline cime_config/allactive/config_pesall.xml:6973]: Maybe adjust these from 72 to 144? GNU is running into OOM for
```
SMS_Ld2.ne30pg2_r05_EC30to60E2r2.BGCEXP_CNTL_CNPECACNT_1850.anvil_gnu.elm-bgcexp
SMS_Ld2.ne30pg2_r05_EC30to60E2r2.BGCEXP_CNTL_CNPRDCTC_1850.anvil_gnu.elm-bgcexp
```

Comment (jonbob) [reply cime_config/allactive/config_pesall.xml:6973]: @amametjanov - what do you think about modifying a new default based on either the ne30pg2_r05_EC30to60E2r2 resolution or the BGC compsets? Or maybe a combination of the two? I'm happy to take that on, but I'd appreciate understanding your preference

Comment (amametjanov) [reply cime_config/allactive/config_pesall.xml:6973]: This layout is for `ne30pg2_EC30to60E2r2` + `WCYCL1850` cases and it also gets picked up for `ne30pg2_r05_EC30to60E2r2` + `BGCEXP_CNTL_CNPECACNT_1850` test-cases.

As you suggest, it's likely better to add a new small node-count default for BGC compsets (and keep this WCYCL layout). Please take a look at #4886, which disambiguates WCYCL (long compset name ends with `SESP` [with EOL regex `$`]) and BGC (long compset name contains `BGC` after `SESP`). It should run on the fewest number of nodes without running into OOM with either intel or gnu compilers.

**Pull request merged**